R/get_zoltar_truth.R

```diff
@@ -0,0 +1,121 @@
+#' Generate versioned truth data for zoltar for a specified target
+#' It only includes national and state-level truth data.
+#' 
+#' @param target string specifying target. 
+#' Currently only support "Cumulative Deaths" and "Incident Deaths".
+#' @param issue_date optional date specifying issue date of truth data.
+#' Default to NULL which will load the latest truth data from covidData. 
+#' 
+#' @return data frame with timezero, unit, target and value
+#'
+#' @export
+preprocess_truth_for_zoltar <- function(target, issue_date = NULL){
+  
+  # validate target 
+  target <- match.arg(target, 
+                      choices = c("Cumulative Deaths", "Incident Deaths"), 
+                      several.ok = FALSE)
+  
+  # load the most up to date weeky truth data from JHU CSSE
+  df <- covidData::load_jhu_data(issue_date = issue_date, 
+                                 spatial_resolution = c('national', 'state'),
+                                 temporal_resolution = 'weekly', 
+                                 measure = 'deaths',
+                                 replace_negatives = FALSE, 
+                                 adjustment_cases = 'none',
+                                 adjustment_method = 'none')
+  
+  # select columns needed for target
+  if (target == "Cumulative Deaths"){
+    target_var <- " wk ahead cum death"
+    df <- df %>%
+      dplyr::select(-inc) %>%
+      dplyr::rename(value = cum, unit = location)
+  } else {
+    target_var <- " wk ahead inc death"
+    df <- df %>%
+      dplyr::select(-cum) %>%
+      dplyr::rename(value = inc, unit = location)
+  }
+  
+  df$horizon <- 20
+  df <- df[rep(row.names(df), df$horizon),]
+  df <- df %>%
+    dplyr::group_by(unit, date, value) %>%
+    # going backward in time
+    dplyr::mutate(horizon = seq(-1,-20)) %>%
+    dplyr::ungroup() %>%
+    dplyr::mutate(
+      # generate the Saturday target end date in a number of weeks behind
+      target_end_date = calc_target_week_end_date(date, horizon),
+      # get the next Monday relative to Saturday target end date
+      timezero = lubridate::ceiling_date(as.Date(target_end_date), 'week') + 1,
```

Comment:
I found the calculation of timezero here confusing.  Is this needed?  If it's not used later in the function, it would probably be clearest to remove its calculation.